### ISSUE TITLE
@rollup/gulp

### ISSUE THREAD

**Issue by waynevanson** [rollup/rollup#4137]
Hi Team,

Thanks for being here. I would love to see a dedicated Rollup plugin for gulp.

If there's a better place to put a request like this, feel free to move it there and advise the outcome.

##### Current

Currently, `@rollup/stream` can't be piped in a stream as it represents a `Readable` stream, and is abstractly the source of files.

Example from `@rollup/stream`.

```ts
import rollupStream from '@rollup/stream';
import gulp from 'gulp';
import terser from 'gulp-terser';
import source from 'vinyl-source-stream';

gulp.task('rollup', () => {
  const options = { input: 'src/index.js' };
  return rollupStream(options)
    .pipe(source('bundle.js'))
    .pipe(terser({ keep_fnames: true, mangle: false }))
    .pipe(gulp.dest('dist'));
});
```
##### After

As a gulp user, this feels delicious and the DX incentivizes users to use `rollup`

```ts
import gulp from "gulp";
import rollup from "@rollup/gulp"

// gulp task
export const build = async () =>
  gulp
    .src("src/index.ts")
    .pipe(rollup({ /* configuration here... */ }))
    .pipe(gulp.dest("dist/"))
```

#### Multiple files

From my understanding of usage, stream, and the implementation of `@rollup/stream`, it cannot handle multiple chunks in a single configuration. `RollupOptions.output.manualChunks` allows multiple chunks to be generated, but will be concatenated together in a single file.

#### Why a plugin?

Why not just use `@rollup/stream`?

`@rollup/stream` also removes the metadata of the file, making it a non-ideal candidate to leverage rolllup's output of multiple chunks.

### Feature Proposal

I've been working on a plugin at https://github.com/waynevanson/rollup-stream-gulp.
Basically, it generates the chunks per `Vinyl.path` using `rollup.generate` and creates a new `Vinyl` per chunk (to which a Vinyl is a file metadata object)

It's a work in progress, but has enough basic support to build itself using gulp.

**Comment by shellscape** [comment]
_(So I jumped the gun on the initial transfer and realized there's no better place to discuss org-wide stuffs :))_

Thanks for the suggestion. I think a third party plugin/wrapper with a bullet on https://github.com/rollup/awesome would be great. While I can see that you're passionate about Gulp, and that's great, I'm in the camp that sees the tool as a relic, superfluous and unneeded complexity to a build pipeline. I'm speaking as someone who migrated from Grunt to Gulp almost immediately when it was released, was a fan for many years, and have authored extensive build pipelines and companion libraries to the tool. I just don't see it as necessary these days, and I think a third party plugin/wrapper is plenty good for the ecosystem.

That's my vote, but others will weigh in I'm sure.

**Comment by waynevanson** [comment]
Thank you for sharing your passion and experience, additional respect granted.

> I'm in the camp that sees the tool as a relic, superfluous and unneeded complexity to a build pipeline

What is the alternative you prefer to use? (off topic)
I actually prefer `AsyncIterator` syntax, so perhaps I'll explore that some more (off topic)

Context on my experience: I haven't been using gulp for long, only a month, but am an advocate of build performance and admire the use of streams. 

I would like the goal of this exchange to be that users of both gulp and rollup know where to look and what to use.
`@rollup/gulp` is a clear winner in this case, I know its officially supported.
When I started using Gulp, I decided to try Rollup inside of it. Even Gulp docs using Rollup were outdated on which packages to import.

Here are the options I see plausible to achieve the goal (no all mutually exclusive):
- Create `@rollup/gulp`
- Update `@rollup/stream` to allow `objectMode: true` so we can inline transform a gulp plugin and allow multiple file outputs
- The bullet on the `rollup/awesome`, as you've suggested.
- Add caveats to `@rollup/stream` indicating lack of multiple bundles per instance
- Update docs at the gulpjs repository

**Comment by StreetStrider** [comment]
I used to use [gulp-better-rollup](https://www.npmjs.com/package/gulp-better-rollup). This one works well, as I remember.

**Comment by waynevanson** [comment]
Peer dependency is rollup 1. Would that be problematic? I think we're on version 2

**Comment by StreetStrider** [comment]
@waynevanson it should be ok, I used this with `rollup@2` and it worked, I got tests that confirm that for my setup.
